### PR TITLE
Add 'this' to some extension methods for net46

### LINKED ISSUE #15156: Purpose of net46 AclExtensions
What's the purpose net46 AclExtension files in CoreFX repo ? They expose different API then their CoreFX/netstandard counterparts and missing quite few checks.

One example is https://github.com/dotnet/corefx/blob/296dc18f32e718280794becd8eccc67aab50251b/src/System.Threading.AccessControl/src/System/Threading/ThreadingAclExtensions.net46.cs

### PR DESCRIPTION
**Pull request by kouvel**
Fixes #15156

### CODE REVIEW

**Comment by kouvel** [comment]
@rahku @gkhanna79

File: src/System.Threading.AccessControl/tests/ThreadingAclExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            var s = new Semaphore(1, 1);
+            var m = new Mutex();
+
+            e.GetAccessControl();
+            e.SetAccessControl(new EventWaitHandleSecurity());
+            s.GetAccessControl();
+            s.SetAccessControl(new SemaphoreSecurity());
+            m.GetAccessControl();
+            m.SetAccessControl(new MutexSecurity());
+        }
```
Comment (stephentoub) [inline src/System.Threading.AccessControl/tests/ThreadingAclExtensionsTests.cs]: Is there anything we can assert about the result of these calls?  At the very least seems like we could Assert.NotNull?

Comment (kouvel) [inline src/System.Threading.AccessControl/tests/ThreadingAclExtensionsTests.cs]: Done

**Comment by stephentoub** [comment]
Thanks

**Pull request merged**